src/Symfony/Component/Security/Core/Encoder/EncoderFactory.php

```diff
@@ -89,7 +106,29 @@ private function getEncoderConfigFromAlgorithm(array $config): array
             // "plaintext" is not listed as any leaked hashes could then be used to authenticate directly
             foreach ([SodiumPasswordEncoder::isSupported() ? 'sodium' : 'native', 'pbkdf2', $config['hash_algorithm']] as $algo) {
                 $config['algorithm'] = $algo;
-                $encoderChain[] = $this->createEncoder($config);
+                $encoderChain[] = $this->createEncoder($config, true);
+            }
+
+            return [
+                'class' => MigratingPasswordEncoder::class,
+                'arguments' => $encoderChain,
+            ];
+        }
+
+        if ($fromEncoders = ($config['migrate_from'] ?? false)) {
+            $encoderChain = [];
+            foreach ($fromEncoders as $name) {
+                if ($encoder = $this->encoders[$name] ?? false) {
+                    $encoder = $encoder instanceof PasswordEncoderInterface ? $encoder : $this->createEncoder($encoder, true);
+                } else {
+                    $encoder = $this->createEncoder(['algorithm' => $name], true);
+                }
+
+                if ($encoder instanceof PlaintextPasswordEncoder) {
+                    throw new LogicException('Migrating from plaintext encoders is not allowed.');
```

Comment:
I'm not sure about this one: that's quite arbitrary - if one does want to migrate from plaintext, better not block them

Reply:
removed